components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsFetchResourceImplTest.java

```diff
@@ -0,0 +1,355 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DefaultPatientsFetchResourceImplTest
+{
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private static final String EID1 = "EID1";
+
+    private static final String EID2 = "EID2";
+
+    private static final String ID_1 = "P0000001";
+
+    private static final String ID_2 = "P0000002";
+
+    private static final String ID_3 = "P0000003";
+
+    private static final String ID_4 = "P0000004";
+
+    @Rule
+    public final MockitoComponentMockingRule<PatientsFetchResource> mocker =
+        new MockitoComponentMockingRule<PatientsFetchResource>(DefaultPatientsFetchResourceImpl.class);
+
+    @Mock
+    private Patient patient1;
+
+    @Mock
+    private Patient patient2;
+
+    @Mock
+    private Patient patient3;
+
+    @Mock
+    private Logger logger;
+
+    private PatientRepository repository;
+
+    private Container container;
+
+    private QueryManager qm;
+
+    private DefaultPatientsFetchResourceImpl component;
+
+    private final DocumentReference patientReference1 = new DocumentReference("wiki", "data", ID_1);
+
+    private final DocumentReference patientReference2 = new DocumentReference("wiki", "data", ID_2);
+
+    private final DocumentReference patientReference3 = new DocumentReference("wiki", "data", ID_1);
+
+    private final JSONObject patient1JSON = new JSONObject().put("id", ID_1);
+
+    private final JSONObject patient2JSON = new JSONObject().put("id", ID_2);
+
+    private final JSONObject patient3JSON = new JSONObject().put("id", ID_3);
+
+    private final Collection<String> uriList = ImmutableList.of("http://uri");
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException
+    {
+        MockitoAnnotations.initMocks(this);
+        final Execution execution = mock(Execution.class);
+        final ExecutionContext executionContext = mock(ExecutionContext.class);
+        final ComponentManager componentManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(componentManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
```

Comment:
The lines from 130 to here aren't needed, they're boilerplate for an older way of getting the `XWikiContext`.